// deck-of-cards/lib/modules/blackjack/modules/hit.js

import getFontSize from 'deck-of-cards/lib/fontSize'
var fontSize

export default {
  deck: function (deck) {
    deck.blackjack.hit = function (next, isPlayer, hand) {
      var cards = deck.cards

      var len = cards.length

      fontSize = getFontSize();
      var card = cards.pop();
      card.blackjack.hit(len, function () {
        card.setSide('front');
        hand.push(card);
        hand.forEach((card, index) => card.animateTo({
          delay: 0,
          duration: 250,
          x: Math.round((index - 2.05 - 1) * 70 * fontSize / 16) -200,
          y: Math.round((isPlayer ? + 150 : -150) * fontSize / 16),
          rot: 0,
        }))
        next(hand);
      }, isPlayer, hand);
    }
  },
  card: function (card) {
    var $el = card.$el

    card.blackjack.hit = function (len, cb, isPlayer, hand) {
      card.animateTo({
        delay: 0,
        duration: 250,
        x: Math.round((hand.length - 2.05) * 70 * fontSize / 16) -200,
        y: Math.round((isPlayer ? + 150 : -150) * fontSize / 16),
        rot: 0,

        onStart: function () {
          $el.style.zIndex = (len - 1)
        },
        onComplete: function () {
          cb()
        }
      })
    }
  }
}
