// deck-of-cards/lib/modules/blackjack/modules/open.js

import getFontSize from 'deck-of-cards/lib/fontSize'

var fontSize

export default {
  deck: function (deck) {
    deck.blackjack.open = function (next, isPlayer) {
      var cards = deck.cards

      var len = cards.length
      const dealt = [];
      fontSize = getFontSize()
      cards.splice(-2, 2).reverse().forEach(function (card, i) {
        card.blackjack.open(i, len, function (i) {
          (!isPlayer && i === 0) ? card.setSide('back') : card.setSide('front');
          dealt.push(card);
          if (i === 1) {
            next(dealt)
          };
        }, isPlayer)
      })
    }
  },
  card: function (card) {
    var $el = card.$el

    card.blackjack.open = function (i, len, cb, isPlayer) {
      var delay = i * 250

      card.animateTo({
        delay: delay,
        duration: 250,
        x: Math.round(((!isPlayer && i === 1) ? -2.05 : i - 2.05) * 70 * fontSize / 16) -200,
        y: Math.round((isPlayer ? +150 : -150) * fontSize / 16),
        rot: (!isPlayer && i === 1) ? 20 : 0,

        onStart: function () {
          $el.style.zIndex = (len - 1) + i
        },
        onComplete: function () {
          cb(i)
        }
      })
    }
  }
}
