// deck-of-cards/lib/modules/poker.js

import getFontSize from '../fontSize'

var fontSize

export default {
  deck: function (deck) {
    // deck.poker = deck.queued(poker)

    deck.poker = poker;

    function poker (next) {
      var cards = deck.cards
      var len = cards.length
      const dealt = [];
      fontSize = getFontSize()
      cards.slice(-2).reverse().forEach(function (card, i) {
        card.poker(i, len, function (i, rank) {
          card.setSide('front')
          dealt.push(rank);
          if (i === 1) {
            next(dealt)
          }
        })
      })
    }
  },
  card: function (card) {
    var $el = card.$el

    card.poker = function (i, len, cb) {
      var delay = i * 250
      

      card.animateTo({
        delay: delay,
        duration: 250,

        x: Math.round((i - 2.05) * 70 * fontSize / 16),
        y: Math.round(-110 * fontSize / 16),
        rot: 0,

        onStart: function () {
          $el.style.zIndex = (len - 1) + i
        },
        onComplete: function () {
          cb(i, card.rank)
        }
      })
    }
  }
}
